We assume that fallocate is supported somehow
but we need to check if we have fallocate()
this problem shows up on uclibc systems since
uclibc does not have fallocate() implemented

Signed-off-by: Khem Raj <raj.khem@gmail.com>

Upstream-Status: Pending

Index: e2fsprogs-1.42/lib/ext2fs/unix_io.c
===================================================================
--- e2fsprogs-1.42.orig/lib/ext2fs/unix_io.c	2012-01-17 17:24:34.290780625 -0800
+++ e2fsprogs-1.42/lib/ext2fs/unix_io.c	2012-01-17 17:25:37.338783680 -0800
@@ -895,7 +895,7 @@
 		goto unimplemented;
 #endif
 	} else {
-#ifdef FALLOC_FL_PUNCH_HOLE
+#if defined FALLOC_FL_PUNCH_HOLE && defined HAVE_FALLOCATE
 		/*
 		 * If we are not on block device, try to use punch hole
 		 * to reclaim free space.
